apps/browser/supabase/migrations/20231206012608_chat_history.sql

```diff
@@ -0,0 +1,83 @@
+drop table roles CASCADE;
+drop table chats CASCADE;
+drop table messages CASCADE;
+
+CREATE TABLE "public"."chats" (
```

Comment:
I think the `goals` table needs to be linked into this new DB structure somehow. Currently we don't have a way of correlating a row in `goals` to a row in `chats`.

Maybe we should `ALTER TABLE "public"."goals" ADD COLUMN chat_id uuid;`

Reply:
i would like to discuss this more in detail, since `goal` was a table for the first iteration (it should probably be gone); we should normalize the database to avoid these kinds of details - and we can get the goals by querying the message users from the chat

Reply:
The "goals" table is currently being used for a few things:
- easy access list of all user-defined goals
- easily understand which requests are subsidized, and how many requests they're using

I think this information is still useful, and we've also built BI dashboards ontop of this table already. So in my opinion even though this table *could* be deprecated in favor of the new table, I think it "doesn't hurt" to keep it around as it continues to provide insights for those dashboards. We can easily link it to the `chats` table so that it's correlated with our other data.

Reply:
Talked with Nestor and he agreed we should be linking the "legacy" `goals` table with the new `chats` table. Made a commit adding this linkage.